client/sites/settings/administration/tools/leave-site/leave-site-modal-form.tsx

```diff
@@ -0,0 +1,134 @@
+import page from '@automattic/calypso-router';
+import {
+	Button,
+	CheckboxControl,
+	Flex,
+	FlexItem,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import QuerySitePurchases from 'calypso/components/data/query-site-purchases';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getCurrentUserId } from 'calypso/state/current-user/selectors';
+import { hasLoadedSitePurchasesFromServer } from 'calypso/state/purchases/selectors';
+import hasCancelableSitePurchases from 'calypso/state/selectors/has-cancelable-site-purchases';
+import { leaveSite } from 'calypso/state/sites/actions';
+import { getSite, getSiteDomain } from 'calypso/state/sites/selectors';
+import LeaveSiteModalWarning from './leave-site-modal-warning';
+import type { AppState } from 'calypso/types';
+import './leave-site-modal-form.scss';
+
+export interface LeaveSiteModalFormProps {
+	siteId: number;
+	onClose?: () => void;
+}
+
+const LeaveSiteModalForm = ( { siteId, onClose }: LeaveSiteModalFormProps ) => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ isChecked, setChecked ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const userId = useSelector( ( state: AppState ) => getCurrentUserId( state ) );
+	const siteDomain = useSelector( ( state: AppState ) => getSiteDomain( state, siteId ) || '' );
+	const siteOwnerId = useSelector( ( state: AppState ) => getSite( state, siteId )?.site_owner );
+	const sitePurchasesLoaded = useSelector( ( state: AppState ) =>
+		hasLoadedSitePurchasesFromServer( state )
+	);
+
+	const hasActiveSubscriptions = useSelector( ( state: AppState ) =>
+		hasCancelableSitePurchases( state, siteId )
+	);
+
+	const isSiteOwner = userId === siteOwnerId;
+
+	const handleLeaveSite = async () => {
+		if ( ! isChecked ) {
+			return;
+		}
+
+		try {
+			setIsSubmitting( true );
+			const result = await dispatch( leaveSite( siteId ) );
+			if ( result ) {
+				page.redirect( '/sites' );
```

Comment:
If I leave a site from the /sites sites list, the action succeeds but the modal is still open 🥲